STYLE_GUIDE.md

```diff
@@ -0,0 +1,481 @@
+```
+
+**Summaries to improve:**
+```yaml
+# Too vague
+summary: "Some thoughts on GraphQL"
+
+# Better
+summary: "Defining a meaningful distinction within code-first GraphQL server libraries."
+
+# Too clickbaity
+summary: "This one weird trick will change how you write TypeScript!"
+
+# Better
+summary: "Using TypeScript's type system to catch bugs at compile time rather than runtime."
+
+# Incomplete sentence
+summary: "Building a Twitter bot and what I learned"
+
+# Better
+summary: "How my Twitter bot works under the hood, how it has evolved over time, and what I've learned from the project."
+```
+
+## Markdown Formatting
+
+### Links
+**Use inline markdown links** for all references:
+```markdown
+[link text](https://example.com)
+```
+
+**Avoid** reference-style links unless you're using the same URL multiple times in the same post.
+
+### Emphasis
+
+#### Italics
+Use **single underscores** for italic text:
+```markdown
+_This is italic text_
+```
+
+**Common uses:**
+- Book, movie, or product names
+- Introducing new terms
+- Attribution at the end of posts: `_Thanks to Person Name for feedback._`
+- Foreign words or phrases
+- Metadata/preamble text: `_Discussed on [Hacker News](...)_`
+
+#### Bold
+Use **double asterisks** for bold text:
+```markdown
+**This is bold text**
+```
+
+**Common uses:**
+- Emphasizing key points within sentences
+- Important concepts or warnings
+- Section of text you want to stand out
+
+#### Code
+Use **backticks** for inline code:
+```markdown
+`someFunction()` or `variableName`
+```
+
+### Code Blocks
+
+Always specify the language for syntax highlighting:
+
+````markdown
+```javascript
+const foo = "bar";
+```
+
+```typescript
+type User = { name: string };
+```
+
+```python
+def hello():
+    return "world"
+```
+````
+
+**Common language identifiers:**
+- `javascript` (preferred over `js`)
+- `typescript` (preferred over `ts`)
+- `jsx`
+- `tsx`
+- `python`
+- `graphql`
+- `bash` or `shell`
+
+### Headers
+
+Use `##` (H2) for main section headers within posts. The post title is H1 and should not be repeated in the body.
+
+**Use sentence case** for headers, same as titles:
+```markdown
+## This is a section header
+## Making it fast and cheap
+## What's the difference?
+```
+
+**Avoid:**
+- Using `#` (H1) in post body
+- Title case for headers: ~~## This Is A Section Header~~
+
+### Horizontal Rules
+
+Use **three hyphens** for horizontal rules:
+```markdown
+---
+```
+
+**Common uses:**
+- Separating preamble/metadata from main content
+- Before attribution/acknowledgments at the end of posts
+- Between major conceptual sections (use sparingly)
+
+### Lists
+
+#### Unordered Lists
+Use **hyphens** for bullet points:
+```markdown
+- First item
+- Second item
+- Third item
+```
+
+#### Ordered Lists
+Use **numbers with periods**:
+```markdown
+1. First step
+2. Second step
+3. Third step
+```
+
+### Block Quotes
+
+Use `>` for block quotes:
```

Comment:
Clarify when to use block quotes vs inline and what formatting to use for attributing quotes.